src/help/git.ts

```diff
@@ -7,17 +7,61 @@ import * as shell from './shell';
  * @param targetDir the clone directory.
  */
 export function clone(repositoryUrl: string, targetDir: string) {
-  const gitHubUseSsh = process.env.GITHUB_USE_SSH;
+  const gitHubUseSsh = detectSSH();
   if (gitHubUseSsh) {
     const sshRepositoryUrl = repositoryUrl.replace('/', ':');
     shell.run(`git clone git@${sshRepositoryUrl}.git ${targetDir}`);
   } else {
-    const gitHubToken = process.env.GITHUB_TOKEN;
+    const gitHubToken = getToken(detectGHE());
     if (!gitHubToken) {
       throw new Error('GITHUB_TOKEN env variable is required when GITHUB_USE_SSH env variable is not used');
     }
     shell.run(`git clone https://${gitHubToken}@${repositoryUrl}.git ${targetDir}`);
+
+  }
+}
+
+/**
+ * Checks if the GitHub API URL set in the GitHub Actions workflow is not the public GitHub API URL.
+ * https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+ *
+ * @return True if GITHUB_API_URL env var is defined and not equal to public github API URL, false,
+ */
+export function detectGHE(): boolean {
```

Comment:
Let's be nice and also check `process.env.GH_HOST` in here:

```
(GH_HOST && GH_HOST != 'github.com') || GITHUB_API_URL && GITHUB_API_URL != 'https://api.github.com'
```

each with the lowercase trim